json/src/de.rs

```diff
@@ -107,6 +108,7 @@ impl<R: Read> DeserializerImpl<R> {
         DeserializerImpl {
             read: read,
             str_buf: Vec::with_capacity(128),
+            remaining_depth: 128,
```

Comment:
How about adding a `Deserializer::with_recursion_limit` method so this can be customized?  Or maybe a `Deserializer` builder with a `with_recursion_limit` method?